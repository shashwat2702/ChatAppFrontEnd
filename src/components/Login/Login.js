import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { withRouter } from 'react-router-dom'
import Input from '../shared/Input/Input';
import Button from '../shared/Button/Button';
import { postData } from '../../utils/getData';
import './Login.scss';

class Login extends Component {
  onSubmit = () => {
    postData('http://localhost:8080/login', {email:'shashwat@gmail.com',password:'Pass@1'})
    .then(({data, status})=> {
      console.log(data,status);
      if(data==='Authenticated') {
        this.props.history.push('/home');
      }
    });
  }
  render() {
    console.log(this.props);
    return (
        <div className="loginForm">
          <ul className="tab-group">
            <li className="tab"><Link to="/registration">Sign Up</Link></li>
            <li className="tab active"><Link to="/">Log In</Link></li>
          </ul>
  
          <div className="tab-content"> 
            <h1>Welcome Back</h1>
            <form
              onSubmit={ (e) => { e.preventDefault(); } }
            >
              <div className="field-wrap">
                <Input
                  id={ 'LoginEmail'}
                  label={'Email ID'}
                  className={'req'}
                  spanContent={'*'}
                  type={'email'}
                  placeholder={'example@mail.com'}
                  autoComplete={'off'}
                />
              </div>
              <div className="field-wrap">
              <Input
                  id={ 'LoginPassword'}
                  label={'Password'}
                  className={'req'}
                  spanContent={'*'}
                  type={'password'}
                  placeholder={'Password'}
                  autoComplete={'off'}
                />
              </div>   
              <Button type="submit" label={'Log In'} onClick={this.onSubmit}/>
            </form>
            <div id="login">   
            </div>
          </div>
        </div>
    )
  }
}
export default withRouter(Login);
