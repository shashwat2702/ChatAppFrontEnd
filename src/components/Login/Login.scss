$body-bg: #c1bdba;
$form-bg: #13232f;
$white: #ffffff;

$main: #1ab188;
$main-light: lighten($main,5%);
$main-dark: darken($main,5%);

$gray-light: #a0b3b0;
$gray: #ddd;

$thin: 300;
$normal: 400;
$bold: 600;
$br: 4px;

*, *:before, *:after {
  box-sizing: border-box;
}

.loginForm {
  background:rgba($form-bg,.9);
  padding: 40px;
  max-width:600px;
  margin:20px auto;
  border-radius:$br;
  box-shadow:0 4px 10px 4px rgba($form-bg,.3);
  a {
    text-decoration:none;
    color:$main;
  }
  .tab-group {
    list-style:none;
    padding:0;
    margin:0 0 40px 0;
    &:after {
      content: "";
      display: table;
      clear: both;
    }
    li a {
      display:block;
      padding:15px;
      background-color:rgba($gray-light,.25);
      color:$gray-light;
      font-size:20px;
      float:left;
      width:50%;
      text-align:center;
      cursor:pointer;
    }
    .active a {
      background:$main;
      color:$white;
    }
  }
  
  .tab-content > div:last-child {
    display:none;
  }
  
  
  h1 {
    text-align:center;
    color:$white;
    font-weight:$thin;
    margin:0 0 40px;
  }
  
    .req {
      margin:2px;
      color:$main;
    }
  
  .field-wrap {
    position:relative;
    margin-bottom:30px;
  }
}


