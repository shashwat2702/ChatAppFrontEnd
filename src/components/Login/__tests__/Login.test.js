import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Login';

jest.mock('react-router-dom');

describe('Login snapshot', () => {
  it('should render correctly', () => {
    const LoginSnap = renderer.create(<Login />);
    const LoginJson = LoginSnap.toJSON();
    expect(LoginJson).toMatchSnapshot();
  });
});
