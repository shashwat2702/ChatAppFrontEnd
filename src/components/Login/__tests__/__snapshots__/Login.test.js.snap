// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot should render correctly 1`] = `
<div
  className="loginForm"
>
  <ul
    class="tab-group"
  >
    <li
      class="tab"
    />
    <li
      class="tab active"
    />
  </ul>
  <div
    class="tab-content"
  >
    <h1>
      Welcome Back
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        class="field-wrap"
      >
        <div
          className="inputComponent"
        >
          <label
            htmlFor="LoginEmail"
          >
            Email ID
            <span
              className="req"
            >
              *
            </span>
          </label>
          <input
            autoComplete="off"
            id="LoginEmail"
            placeholder="example@mail.com"
            type="email"
          />
        </div>
      </div>
      <div
        class="field-wrap"
      >
        <div
          className="inputComponent"
        >
          <label
            htmlFor="LoginPassword"
          >
            Password
            <span
              className="req"
            >
              *
            </span>
          </label>
          <input
            autoComplete="off"
            id="LoginPassword"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
      <div
        className="buttonComponent"
      >
        <button
          className="button button-block"
          onClick={[Function]}
          type="submit"
        >
          Log In
        </button>
      </div>
    </form>
    <div
      id="login"
    />
  </div>
</div>
`;
